Vladimir Ivanov SI/Pract1.4/Pract1.4.cpp

```diff
@@ -0,0 +1,115 @@
+#include <iostream>
+using std::cin;
+using std::cout;
+using std::endl;
+
+const unsigned short NAME_MAX_SIZE = 100;
+const unsigned short FN_MAX_SIZE = 10;
+double groupAverageGrade = 0;
+
+struct Student {
+    char name[NAME_MAX_SIZE];
+    char fn[FN_MAX_SIZE];
+    double grade;
+};
+
+void readStudents(const unsigned int studentsCount, Student*& students) {
+    for (unsigned int i = 0; i < studentsCount; i++) {
+        cin >> students[i].name >> students[i].fn >> students[i].grade;
+        groupAverageGrade += students->grade;
+    }
+
+    groupAverageGrade /= studentsCount;
+}
+
+unsigned int chechStudentShip(const double studentShip, const Student* students, const unsigned int studentsCount) {
+    unsigned int studentsWithShip = 0;
+    for (unsigned int i = 0; i < studentsCount; i++) {
+        if (students[i].grade >= studentShip) {
+            studentsWithShip++;
+        }
+    }
+
+    return studentsWithShip;
+}
+
+void fillSortedStudentsWithShips(const double studentShip, const unsigned int studentsCount, Student* sortedStudentsWithShips, const Student* students) {
+    unsigned int k = 0;
+    for (unsigned int i = 0; i < studentsCount; i++) {
+        if (students[i].grade >= studentShip) {
+            sortedStudentsWithShips[k++] = students[i];
+        }
+    }
+}
+
+void swapStudents(Student*& sortedStudentsWithShips, const unsigned int j) {
+    Student tempStudent = sortedStudentsWithShips[j];
+    sortedStudentsWithShips[j] = sortedStudentsWithShips[j + 1];
+    sortedStudentsWithShips[j + 1] = tempStudent;
+}
```

Comment:
This is not a very abstract swap function, you need to pass the two positions that will be swapped